Sources/TensorFlow/Core/VectorProtocol.swift

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import _Differentiation
+
+#if TENSORFLOW_USE_STANDARD_TOOLCHAIN
+@_spi(Reflection) import Swift
+
+public protocol _VectorProtocol {
+  typealias VectorSpaceScalar = Float
```

Comment:
Do you mean `associatedtype`?

Reply:
_VectorProtocol cannot have any associated type or Self constrained that would prevent doing type as? _VectorProtocol.Type.

Reply:
Oh I see. So this only works with Float now? That seems fine to me. `VectorProtocol` isn't very good anyways and we should stop using it eventuallly.

Does this make the `associatedtype VectorSpaceScalar` inside of `VectorProtocol` irrelevant? Can you remove it so that it's not confusing?

Reply:
Cannot. It is needed to be associatedtype so it can be in the where clauses until we can delete them.

Reply:
Oh, ok